src/routing_core.rs

```diff
@@ -743,9 +743,313 @@ impl RoutingCore {
+                                        },
+                                        &ExpectedConnection::Response(ref response, _) => {
+                                            // We initiated the ConnectRequest, node A on diagram
+                                            // RFC-0011.
+                                            if request.requester_fob.name() == self.id().name() &&
+                                                    hello.public_id == response.receiver_fob {
+                                                match value.0 {
+                                                    Some(connection) => {
+                                                        opt_connection = Some(connection.clone());
+                                                        break;
+                                                    },
+                                                    None => {},
+                                                }
+                                            }
+                                        },
+                                    }
+                                }
+                            },
+                            ExpectedConnection::Response(ref response, _) => {
+                                for (key, value) in self.expected_connections.iter() {
+                                    match key {
+                                        &ExpectedConnection::Request(ref request) => {
+                                            // We are the network-side with a ConnectRequest, Node B
+                                            // on diagram of RFC-0011.
+                                            if response.receiver_fob.name() == self.id().name() &&
+                                                    hello.public_id == request.requester_fob {
+                                                match value.0 {
+                                                    Some(connection) => {
+                                                        opt_connection =
+                                                            Some(unknown_connection.clone());
+                                                        break;
+                                                    },
+                                                    None => {},
+                                                }
+                                            }
+                                        },
+                                        &ExpectedConnection::Response(_, _) => {
+                                            // Don't match on ConnectResponse.
+                                        },
+                                    }
+                                }
+                            },
+                        }
+                    },
+                    None => {},
+                };
+
+                match opt_connection {
+                    Some(connection) => {
```

Comment:
but here we need to determine whether we are a client bootstrapping.

Alternatively, and as suggested by Brian, we need to earlier split the program flow and handle as a client before already